sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java

```diff
@@ -0,0 +1,435 @@
+            return withContext(context -> this.uploadManifestWithResponse(ByteBuffer.wrap(bytes), context)).flatMap(FluxUtil::toMono);
+        } catch (IOException exception) {
+            return monoError(logger, new UncheckedIOException(exception));
+        }
+    }
+
+    /**
+     * Uploads a manifest to the repository.
+     * The client currently only supports uploading OciManifests to the repository.
+     * And this operation makes the assumption that the data provided is a valid OCI manifest.
+     * <p>
+     * Also, the data is read into memory and then an upload operation is performed as a single operation.
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     * @param data The manifest that needs to be uploaded.
+     * @return operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<UploadManifestResult> uploadManifest(BinaryData data) {
+        if (data == null) {
+            return monoError(logger, new NullPointerException("'data' can't be null."));
+        }
+
+        return withContext(context -> this.uploadManifestWithResponse(data.toByteBuffer(), context)).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Uploads a manifest to the repository.
+     * The client currently only supports uploading OciManifests to the repository.
+     * And this operation makes the assumption that the data provided is a valid OCI manifest.
+     * <p>
+     * Also, the data is read into memory and then an upload operation is performed as a single operation.
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param data The manifest that needs to be uploaded.
+     * @return The rest response containing the operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<UploadManifestResult>> uploadManifestWithResponse(BinaryData data) {
+        if (data == null) {
+            return monoError(logger, new NullPointerException("'data' can't be null."));
+        }
+
+        return withContext(context -> this.uploadManifestWithResponse(data.toByteBuffer(), context));
+    }
+
+    Mono<Response<UploadManifestResult>> uploadManifestWithResponse(ByteBuffer data, Context context) {
+        if (data == null) {
+            return monoError(logger, new NullPointerException("'data' can't be null."));
+        }
+
+        String digest = UtilsImpl.computeDigest(data);
```

Comment:
Random question, is the digest always needed or can it be optional?

Reply:
As of now it is always needed. 
I have a follow-up bug to review with FC to see whether adding a digest here is worth the performance hit as we have no option but to load everything in memory.